contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -928,6 +929,177 @@ checksum(PG_FUNCTION_ARGS)
+	truncate_tsql_identifier(db_name);
+	truncate_tsql_identifier(schema_name);
+	truncate_tsql_identifier(object_name);
+
+	if (!strcmp(db_name, ""))
+		db_name = get_cur_db_name();
+
+	/* check if looking in current database, if not then return null value */
+	if (strcmp(db_name, get_cur_db_name()) && strcmp(db_name, "tempdb"))
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_DATABASE),
+				 errmsg("Can only do lookup in current database.")));
+
+	if (!strcmp(schema_name, ""))
+	{	
+		/* find the default schema for current user */
+		const char *user = get_user_for_database(db_name);
+		schema_name = get_authid_user_ext_schema_name((const char *) db_name, user);
+	}
+
+	/* get physical schema name from logical schema name */
+	physical_schema_name = get_physical_schema_name(db_name, schema_name);
+
+	/* get schema oid from physical schema name */
+	schema_oid = LookupExplicitNamespace(physical_schema_name, true);
+
+	/* free unnecessary pointers */
+	pfree(db_name);
+	pfree(schema_name);
+	pfree(physical_schema_name);
+
+	if (!OidIsValid(schema_oid))
+	{
+		pfree(object_name);
+		pfree(object_type);
+		PG_RETURN_NULL();
+	}
+	
+	/* check if looking for temp object */
+	is_temp_object = (object_name[0] == '#'? true: false);
+
+	if (strcmp(object_type, "")) /* object_type is not empty */
+	{
+		if (!strcmp(object_type, "S") || !strcmp(object_type, "U") || !strcmp(object_type, "V") ||
+			!strcmp(object_type, "IT") || !strcmp(object_type, "ET") || !strcmp(object_type, "SO"))
+		{
+			if (is_temp_object)
+			{
+				/* search in list of ENRs registered in the current query environment by name */
+				result = tsql_get_temp_object_oid(object_name);
+			}
+			else
+			{
+				/* search in pg_class by name and schema oid */
+				result = get_relname_relid((const char *) object_name, schema_oid); 
+			}
+		}
+		else if (!strcmp(object_type, "C") || !strcmp(object_type, "D") || !strcmp(object_type, "F") ||
+				!strcmp(object_type, "PK") || !strcmp(object_type, "UQ"))
+		{
+			/* search in pg_constraint by name and schema oid */
+			result = tsql_get_constraint_oid(object_name, schema_oid);
+		}
+		else if (!strcmp(object_type, "AF") || !strcmp(object_type, "FN") || !strcmp(object_type, "FS") ||
+				!strcmp(object_type, "FT") || !strcmp(object_type, "IF") || !strcmp(object_type, "P") ||
+				!strcmp(object_type, "PC") || !strcmp(object_type, "TF") || !strcmp(object_type, "RF") ||
+				!strcmp(object_type, "X"))
+		{
+			/* search in pg_proc by name and schema oid */
+			result = tsql_get_proc_oid(object_name, schema_oid);
+		}
+		else if (!strcmp(object_type, "TR") || !strcmp(object_type, "TA"))
+		{
+			/* search in pg_trigger by name */
+			result = tsql_get_trigger_oid(object_name);
+		}
+		else if (!strcmp(object_type, "R") || !strcmp(object_type, "EC") || !strcmp(object_type, "PG") ||
+				!strcmp(object_type, "SN") || !strcmp(object_type, "SQ") || !strcmp(object_type, "TT"))
+		{
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
```

Comment:
Use ERRCODE_FEATURE_NOT_SUPPORTED with "Object type currently unsupported in Babelfish"

Reply:
addressed in the next commit